rfcs/rfc_001.md

```diff
@@ -0,0 +1,161 @@
+# RFC 001 - simplification of Ngeo buildtools and custom frontends for GeoMapfish
+
+## History
+
+Following POC https://github.com/camptocamp/ngeo/pull/9353 this RFC has been reworked.
+
+## Motivation
+
+- Keep Ngeo-based Geomapfish solutions maintainable;
+- Allow creating Ngeo-base Geomapfish solutions with any modern buildtool / framework;
+- Allow alternative frontends in Geomapfish (geogirafe, masterportal, uzuverse, ...);
+
+## Current situation / problems
+
+When scaffolding a Geomapfish project, the Ngeo/Gmf build tools and default interfaces are copied.
+Ngeo/Gmf is added as a dependency and directly consummed as source code.
+There is no alternative frontend, no possibility to mix different versions or use other build tools.
+Notably, the code requires specific transformations, managed by webpack 4 plugins:
+- AngularJS dependency injection with ngInject;
+- SaaS files importation, unicity and ordering;
+- SVG files importation and build-time modification;
+- AngularJS partials importation.
+
+## Vision
+
+### Allow alternative frontends (GeoGirafe, MasterPortal, Uzuverse, other)
+
+- No change in the way geomapfish is deployed;
+- No change in the backend protocol;
+- integrate seemlesly with the existing backend and deployment mechanism;
+- Allow completly independant frontend (GeoGirafe, MasterPortal, Uzuverse, other);
+
+### Simplify Ngeo/Gmf build tools and make the code standard
+
+- Limit breaking changes for existing users;
+- Remove / simplify code;
+- Distribute the Ngeo/GMF library as pure ES6 + CSS package;
+
+### Allow to easily extend Ngeo/Gmf
+
+- Ability to use any build system, frontend framework, or package and in any version;
+- Expose some states on a case-by-case;
+- Document and demo this opt-in mechanism;
+
+
+## Technical proposition part 1: allow alternative frontends
+
+### Creation of a custom frontend
+
+There will be a `frontend` directory at the root of the Geomapfish project. This directory will contain the custom frontend project.
+
+In this folder the developer should add everything needed to build the frontend like a `package.json`, the custom components, the build tools, the interface html files, etc.
+
+This folder will contain a `Dockerfile`, in charge of building this custom frontend project.
+The content is up to the developer, the only constraint is to put the generated files in the `/etc/static-frontend` directory.
```

Comment:
Why this constraint? 
If the image content is up to the developer, then we just have to build an image than runs and can be integrated in GMF, or am I missing something here?